lte/gateway/python/magma/mobilityd/mobility_store.py

```diff
@@ -12,20 +12,55 @@
 """
 from collections import defaultdict
 
-from magma.common.redis.containers import RedisFlatDict, RedisHashDict, \
-    RedisSet
+import redis
+from magma.common.redis.containers import RedisFlatDict, RedisSet
 from magma.common.redis.serializers import RedisSerde
-from magma.mobilityd import serialize_utils
 from magma.common.redis.containers import RedisHashDict
 from magma.common.redis.serializers import get_json_serializer, \
     get_json_deserializer
 from magma.common.redis.client import get_default_client
 
+from magma.mobilityd import serialize_utils
+from magma.mobilityd.ip_descriptor import IPDesc
+from magma.mobilityd.ip_descriptor_map import IpDescriptorMap
+from magma.mobilityd.uplink_gw import UplinkGatewayInfo
+
 IPDESC_REDIS_TYPE = "mobilityd_ipdesc_record"
 IPSTATES_REDIS_TYPE = "mobilityd:ip_states:{}"
 IPBLOCKS_REDIS_TYPE = "mobilityd:assigned_ip_blocks"
 MAC_TO_IP_REDIS_TYPE = "mobilityd_mac_to_ip"
 DHCP_GW_INFO_REDIS_TYPE = "mobilityd_gw_info"
+ALLOCATED_IID_REDIS_TYPE = "mobilityd_allocated_iid"
+ALLOCATED_SESSION_PREFIX_TYPE = "mobilityd_allocated_session_prefix"
+
+
+class MobilityStore(object):
```

Comment:
I like what you've done here, but unless you feel like making this the single go-to storage interface/DAO for all of mobilityd (tbh not even a bad idea), I would just recast this as a flat factory function that returns all the different things that clients need to interact with for state storage. This should only be called in main and potentially unit tests anyways

I think there's some good value in creating an interface that specifies exactly what the contract is for interaction with state storage across the entire service (see our pattern for this on the cloud side e.g. in smsd, configurator, etc), but probably out of reasonable scope for this change. There are too many random state stores floating around, it would be nice to unify all the functionality into a single storage object. Let's make this a bootcamp task or something.

Reply:
I still see value maintaining this class as it is, I agree we should go for simplicity instantiating each field but I do like the wrapping benefits we get by putting this into the "interface" store class, we can define it as the main source of truth for mobilityd and I can add a TODO to later change it into the unified storage object.